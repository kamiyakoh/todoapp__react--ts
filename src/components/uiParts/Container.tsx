import type { ReactNode, FC } from 'react';
import type { SerializedStyles } from '@emotion/react';
/** @jsxRuntime classic */
/** @jsx jsx */
import { css, jsx } from '@emotion/react';

interface Props {
  cssName?: SerializedStyles | SerializedStyles[];
  isSingle: boolean;
  children: ReactNode;
}

export const Container: FC<Props> = ({ cssName, isSingle, children }) => {
  return <div css={[container, cssName, isSingle ? single : multi]}>{children}</div>;
};

const container = css`
  width: 90%;
  margin: 0 auto;
`;
const single = css`
  max-width: 720px;
`;
const multi = css`
  max-width: 1280px;
`;
