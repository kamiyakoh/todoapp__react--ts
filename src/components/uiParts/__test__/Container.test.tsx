import type { SerializedStyles } from '@emotion/react';
/** @jsxRuntime classic */
/** @jsx jsx */
import { css, jsx } from '@emotion/react';
import { render } from '@testing-library/react';
import { Container } from '../Container';

const padding = css`
  padding: 80px;
`;

const minHeight = css`
  min-height: 100vh;
`;

describe('Container Test', () => {
  test.each`
    isSingle
    ${true}
    ${false}
  `('isSingle-Props Test', (isSingle: boolean) => {
    const { asFragment } = render(<Container isSingle={isSingle}>children</Container>);
    expect(asFragment()).toMatchSnapshot();
  });

  test.each`
    cssName
    ${undefined}
    ${[padding, minHeight]}
  `('addClassName Test', (cssName: SerializedStyles | SerializedStyles[]) => {
    const { asFragment } = render(
      <Container isSingle={true} cssName={cssName}>
        children
      </Container>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
