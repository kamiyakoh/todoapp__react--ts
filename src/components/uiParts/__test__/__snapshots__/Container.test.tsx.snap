// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Test addClassName Test 1`] = `
<DocumentFragment>
  <div
    class="css-16vsmse-Container"
  >
    children
  </div>
</DocumentFragment>
`;

exports[`Container Test addClassName Test 2`] = `
<DocumentFragment>
  <div
    class="css-1bh0wdd-Container"
  >
    children
  </div>
</DocumentFragment>
`;

exports[`Container Test isSingle-Props Test 1`] = `
<DocumentFragment>
  <div
    class="css-16vsmse-Container"
  >
    children
  </div>
</DocumentFragment>
`;

exports[`Container Test isSingle-Props Test 2`] = `
<DocumentFragment>
  <div
    class="css-16vsmse-Container"
  >
    children
  </div>
</DocumentFragment>
`;
